src/Streamly/Streams/StreamK.hs

```diff
@@ -912,7 +908,7 @@ takeWhile p m = fromStream $ go (toStream m)
                        | otherwise = stp
             yieldk a r | p a       = yld a (go r)
                        | otherwise = stp
-         in unStream m1 (rstState st) stp single yieldk
+         in unStream m1 st stp single yieldk
 
 drop :: IsStream t => Int -> t m a -> t m a
 drop n m = fromStream $ Stream $ \st stp sng yld ->
```

Comment:
missed removing the "rstState" below?

Reply:
Oh yeah, seemed to have missed it. Should use `unStreamShared` in yieldk instead.